### PR TITLE
Relax numpy upper bound and bump deepspeed version

### PR DESCRIPTION
**Pull request by hvaara**
DeepSpeed `v0.14.4` and later supports current numpy (confirmed in https://github.com/microsoft/DeepSpeed/issues/5671).

This should allow us to get rid of the upper bound on numpy after a version bump of DeepSpeed.

See https://github.com/huggingface/trl/pull/1747 for original pin.

### CODE REVIEW

**Comment by hvaara** [comment]
Running tests locally passed

```
$ make test
212 passed, 180 skipped, 592 warnings in 480.39s (0:08:00)
```

I'm not sure if this actually tests anything related to DeepSpeed with `numpy>=2.0.0`. Will the DeepSpeed integration be tested in CI?

File: setup.py [hunk 1]
```diff
@@ -63,7 +63,7 @@
 REQUIRED_PKGS = [
     "torch>=1.4.0",
     "transformers>=4.31.0",
-    "numpy>=1.18.2,<2.0.0",
+    "numpy>=1.18.2",
```
Comment (lewtun) [inline setup.py:66]: To be safe, let's add the upper bound for the next major release:
```suggestion
    "numpy>=1.18.2,<3.0.0",
```

**Review by lewtun — APPROVED** [review]
Thanks for fixing the deps @hvaara ! LGTM with a small change

File: setup.py (hunk 1 above)
Comment (hvaara) [reply setup.py:66]: Are you sure you really want this? There hasn't been any any known issues with this version yet. By this logic we should add upper bound pins on all the other deps too?

I'll make the change if you really want it. Just want to make sure you do first.

Comment (lewtun) [reply setup.py:66]: I took a look at our other libs and indeed we don't upper bound `numpy`, so let's keep it as is. Thanks!

**Comment by hvaara** [comment]
Thanks for the review! 🤗

**Pull request merged**